src/lib/rosetta_lib/user_command_info.ml

```diff
@@ -355,20 +355,20 @@ let to_operations ~failure_status (t : Partial.t) : Operation.t list =
      * transfer. ie. Source decreases, and receiver increases.
   *)
   let plan : 'a Op.t list =
-    (* The dec side of a user command's fee transfer is here *)
-    (* TODO: Relate fee_payer_dec here with fee_receiver_inc in internal commands *)
-    ( if not Unsigned.UInt64.(equal t.fee zero) then
-      [{Op.label= `Fee_payer_dec; related_to= None}]
-    else [] )
-    @ ( match failure_status with
-      | Some (`Applied (Account_creation_fees_paid.By_receiver amount)) ->
-          [ { Op.label= `Account_creation_fee_via_payment amount
-            ; related_to= None } ]
-      | Some (`Applied (Account_creation_fees_paid.By_fee_payer amount)) ->
-          [ { Op.label= `Account_creation_fee_via_fee_payer amount
-            ; related_to= None } ]
-      | _ ->
-          [] )
+    ( match failure_status with
+    | Some (`Applied (Account_creation_fees_paid.By_receiver amount)) ->
+        [{Op.label= `Account_creation_fee_via_payment amount; related_to= None}]
+    | Some (`Applied (Account_creation_fees_paid.By_fee_payer amount)) ->
+        [ { Op.label= `Account_creation_fee_via_fee_payer amount
+          ; related_to= None } ]
+    | None ->
+        (* mempool *)
+        (* The dec side of a user command's fee transfer is here ONLY when requested from mempool. When loaded from a block, we pull the fee_payer_dec from one side of side of a fee_transfer *)
```

Comment:
Why is this not required for user commands in a block? We will lose the account information for `Fee_payer_dec` corresponding to the `Fee_receiver_inc` of fee transfers (See the above comment)

Reply:
> We will lose the account information for Fee_payer_dec corresponding to the Fee_receiver_inc of fee transfers (See the above comment)

Yes, but this is okay. I think it's better that we more closely reflect on how the protocol represents the fee transfers.

Reply:
But it is wrong to have the same account for both `Fee_payer_dec`  and  `Fee_receiver_inc` for a fee transfer. This basically means the fee transfer did not happen?

Reply:
Talked to deepthi -- I was mistaken because my tests weren't powerful enough. The correct behavior is:
1. Fee payer dec happens when user command is processed
2. Fee payer inc (only!) happens when fee-transfer is processed
3. There is a special fee-transfer that is extracted from a coinbase where fee-payer-dec also happens to the coinbase receiver